go/pkg/array/reduce_test.go

```
@@ -0,0 +1,546 @@
+	numbers := []int{1, 2, 3, 4, 5}
+	sum := Reduce(numbers, func(acc, val int) int {
+		return acc + val
+	}, 0)
+
+	fmt.Printf("Sum: %d\n", sum)
+	// Output: Sum: 15
+}
+
+func ExampleReduce_findMax() {
+	values := []float64{3.14, 2.71, 4.67, 1.41}
+	max := Reduce(values, func(acc, val float64) float64 {
```

golangci-lint (2.2.2)

variable max has same name as predeclared identifier

(predeclared)